Controllers/LeaderboardController.cs

```diff
@@ -300,6 +301,75 @@ public class LeaderboardController : Controller {
                 foreach (var score in leaderboard.Scores) {
                     score.Player = PostProcessSettings(score.Player);
                 }
+
+                // TODO: IS THIS THE RIGHT WAY TO DO THIS?
+                leaderboard.ClanRanking = currentContext
+                .ClanRanking
+                .Include(cr => cr.Clan)
+                .Where(cr => cr.LeaderboardId == leaderboard.Id)
+                .OrderByDescending(cr => cr.ClanPP)
+                .Skip((page - 1) * count)
+                .Take(count)
+                .Select(cr => new ClanRankingResponse
+                {
+                    Id = cr.Id,
+                    Clan = cr.Clan,
+                    LastUpdateTime = cr.LastUpdateTime,
+                    ClanRank = cr.ClanRank,
+                    ClanAverageRank = cr.ClanAverageRank,
+                    ClanPP = cr.ClanPP,
+                    ClanAverageAccuracy = cr.ClanAverageAccuracy,
+                    ClanTotalScore = cr.ClanTotalScore,
+                    LeaderboardId = cr.LeaderboardId,
+                    Leaderboard = cr.Leaderboard,
+                    AssociatedScores = currentContext
+                        .Scores
+                        .Where(s => s.LeaderboardId == leaderboard.Id && s.Player.Clans.Contains(cr.Clan))
+                        .Include(sc => sc.Player)
+                        .ThenInclude(p => p.ProfileSettings)
+                        .Include(s => s.Player)
+                        .ThenInclude(s => s.Clans)
+                        .OrderBy(s => s.Rank)
+                        .Skip((page - 1) * count)
```

Comment:
This is probably incorrect as you use the same page for clan ranking as for scores. Not sure though, will need to check the UI first

Reply:
You're right. I think this is a case I missed on the website. If a clan has a large number of associated scores what should that look like on the leaderboard page on the website? Should there be some [1] [2] [3] page selectors somewhere?

Reply:
I would actually recommend dividing this into a new endpoint and to fetch scores only when the spoiler is open. How score stats is fetched for example

Reply:
Fixed